Lib/multiprocessing/pool.py

```diff
@@ -452,18 +486,30 @@ def _map_async(self, func, iterable, mapper, chunksize=None, callback=None,
         return result
 
     @staticmethod
-    def _handle_workers(cache, taskqueue, ctx, Process, processes, pool,
-                        inqueue, outqueue, initializer, initargs,
-                        maxtasksperchild, wrap_exception):
+    def _wait_for_updates(sentinels, change_notifier, timeout=None):
+        wait(sentinels, timeout=timeout)
+        while not change_notifier.empty():
+            change_notifier.get()
+
+    @classmethod
+    def _handle_workers(cls, cache, taskqueue, ctx, Process, processes,
+                        pool, inqueue, outqueue, initializer, initargs,
+                        maxtasksperchild, wrap_exception, sentinels,
+                        change_notifier):
         thread = threading.current_thread()
 
         # Keep maintaining workers until the cache gets drained, unless the pool
         # is terminated.
         while thread._state == RUN or (cache and thread._state != TERMINATE):
-            Pool._maintain_pool(ctx, Process, processes, pool, inqueue,
-                                outqueue, initializer, initargs,
-                                maxtasksperchild, wrap_exception)
-            time.sleep(0.1)
+            cls._maintain_pool(ctx, Process, processes, pool, inqueue,
+                               outqueue, initializer, initargs,
+                               maxtasksperchild, wrap_exception)
+
+            worker_sentinels = [worker.sentinel for worker in
+                                pool if hasattr(worker, "sentinel")]
```

Comment:
Same here: how about a dedicated method to get worker sentinels that you can override in `ThreadPool`?

Reply:
The problem here is that we don't have the reference to the pool object (`self`) after https://bugs.python.org/issue35378 so we can only call class/static methods. The most we can do here is make a static/class method that takes the list of workers (called `pool` here) and then it returns the list of sentinels.

What do you think?

Reply:
That sounds fine to me.